### PR TITLE
Bump samtools to v18

### PR DESCRIPTION
**Pull request by nvnieuwk**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Review by FriederikeHanssen — COMMENTED** [review]
any chance you can also add all the mulled containers? 🙈

File: modules/nf-core/gatk4/markduplicates/environment.yml [hunk 1]
```diff
@@ -5,4 +5,4 @@ channels:
   - defaults
 dependencies:
   - bioconda::gatk4=4.4.0.0
-  - bioconda::samtools=1.17
+  - bioconda::samtools=1.18
```
Comment (FriederikeHanssen) [inline modules/nf-core/gatk4/markduplicates/environment.yml:8]: this one here should need a mulled bump then as well. I think either update the container or leave this one out

Comment (nvnieuwk) [reply modules/nf-core/gatk4/markduplicates/environment.yml:8]: Oh good spot, thanks I'll leave that one out!

**Comment by FriederikeHanssen** [comment]
> > any chance you can also add all the mulled containers? 🙈
> 
> If I really have to 😉

pleeeeeease 🙏 No I know how much work it is and what a rabbit hole. Can also leave this to someone else ;)

**Comment by nvnieuwk** [comment]
> > > any chance you can also add all the mulled containers? 🙈
> > 
> > 
> > If I really have to 😉
> 
> pleeeeeease 🙏 No I know how much work it is and what a rabbit hole. Can also leave this to someone else ;)

I'll take a look later how much work it would be :)

**Comment by nvnieuwk** [comment]
PR for mulled containers: https://github.com/BioContainers/multi-package-containers/pull/2854

**Comment by FriederikeHanssen** [comment]
For the mapping mulled containers: there seem to be several bug fixes for dealing with CRAm files and sorting: https://github.com/samtools/samtools/releases/

**Comment by nvnieuwk** [comment]
> Hi. By the way, the samtools team are planning the version 1.19 release in the next two weeks. Given the size of the effort, maybe this PR could wait a bit ?

Hi thanks for the heads up, It would really be a whole other effort to go through the whole process again (which I personally don't want to do again) This PR is almost ready now so I don't think I'll wait for that

**Comment by nvnieuwk** [comment]
@FriederikeHanssen @mashehu This one should now finally be ready, can you take a look (I'm sorry for how big this became, had to rewrite half the tests :cry:)

File: tests/subworkflows/nf-core/bam_markduplicates_samtools/main.nf [hunk 2]
```diff
@@ -8,5 +8,5 @@ workflow test_bam_markduplicates_samtools {
 
     input = [ [ id:'test' ], file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true) ]
 
-    BAM_MARKDUPLICATES_SAMTOOLS ( input, [] )
+    BAM_MARKDUPLICATES_SAMTOOLS ( input, [[]] )
```
Comment (nvnieuwk) [inline tests/subworkflows/nf-core/bam_markduplicates_samtools/main.nf:11]:
```suggestion
    BAM_MARKDUPLICATES_SAMTOOLS ( input, Channel.value([]) )
```

**Comment by SPPearce** [comment]
I think those leftover build files are breaking pytest in general.

**Review by mashehu — APPROVED** [review]

**Pull request merged**